pass.go

```diff
@@ -0,0 +1,1054 @@
+// only called by RenderPage, do not call directly
+func (p *Pass) renderPageJSON(rw http.ResponseWriter, r *http.Request, innerTemplate *template.Template, tv interface{}) {
+	rw.Header().Add("Content-Type", "application/json")
+	if b, err := json.MarshalIndent(tv, "", "  "); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		log.Printf("renderPageJSON err %v\n", err)
+	} else {
+		rw.Write(b)
+	}
+}
+
+// RenderPage renders the page using the configured layout and security headers
+func (p *Pass) RenderPage(rw http.ResponseWriter, r *http.Request, innerTemplate *template.Template, tv interface{}) {
+	p.OutputHeaders(rw, r)
+
+	if helpers.HTTPAcceptCheck("application/json", r.Header) {
+		if r.Method != http.MethodPost {
+			p.RenderErrorPage(rw, r, tv, NewHTTPError(405, fmt.Errorf("application/json only accepted for HTTP POST")))
+			return
+		}
+		p.renderPageJSON(rw, r, innerTemplate, tv)
+		return
+	}
+
+	btv := &baseTemplateVariables{
+		PageURL:           p.GetURL(),
+		PageTitle:         p.sv.cfg.Title,
+		CSSIntegrityHash:  p.sv.integrityHashes["css"],
+		JSIntegrityHash:   p.sv.integrityHashes["js"],
+		TemplateVariables: tv,
+	}
+
+	if templateError := innerTemplate.ExecuteTemplate(rw, "base", btv); templateError != nil {
+		log.Printf("Template did not render: %v", templateError)
+
+		// Assume templating is broken, output error string raw to client
+		rw.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintln(rw, "Unexpected error while rendering page!")
+	}
+}
+
+// renderErrorPageJSON does the same as RenderErrorPage, but for json requests
+// only called by RenderErrorPage, do not call directly
+func (p *Pass) renderErrorPageJSON(rw http.ResponseWriter, r *http.Request, errStr string) {
+	rw.Header().Add("Content-Type", "application/json")
+	errorJSON := struct{ Error string }{
+		Error: errStr,
+	}
+
+	if b, err := json.MarshalIndent(errorJSON, "", "  "); err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		log.Printf("renderErrorPageJSON err %v\n", err)
+	} else {
+		rw.Write(b)
+	}
+}
+
+// RenderErrorPage outputs a rendered error template, no output must be made
+// after a call to this function!
+func (p *Pass) RenderErrorPage(rw http.ResponseWriter, r *http.Request, tv interface{}, err error) {
+	var httpStatusCode int
+	var errStr string
+	var innerTemplate *template.Template
+	httpStatusCode, innerTemplate, errStr = p.serverError(err)
+
+	rw.WriteHeader(httpStatusCode)
+
+	if helpers.HTTPAcceptCheck("application/json", r.Header) {
+		p.renderErrorPageJSON(rw, r, errStr)
+		return
+	}
+
+	tv2 := tv.(*TemplateVariables)
+	tv2.ErrorString = errStr
+
+	p.RenderPage(rw, r, innerTemplate, tv2)
+}
+
+// GetURL returns a https link for the configured domain[:port]
+func (p *Pass) GetURL() string {
+	return fmt.Sprintf("https://%s", p.sv.cfg.Hostname)
+}
+
+// GetPassword returns the password stored at the given UUID if any
+func (p *Pass) GetPassword(uuid string) (payload string, err error) {
```

Comment:
`func (p *Pass) GetPassword(uuid string) (string, error) {`

Reply:
Fixed in next push